### PR TITLE
Add support for info

### PR DESCRIPTION
**Pull request by rejuvyesh**
Not sure if this is the best way. Looking for suggestions.

### CODE REVIEW

File: src/DeepRL.jl [hunk 1]
```diff
@@ -25,8 +25,13 @@ export
     sample_action,
     n_actions,
     obs_dimensions,
-    render
+    render,
+    generate_info
 
+# XXX
+function generate_info(p::Union{MDP,POMDP}, s::S, a::A, sp::S) where {S,A}
```
Comment (zsunberg) [inline src/DeepRL.jl:33]: In my experience, it's better to just leave out the type parameters in functions like these. Just use
```julia
function generate_info(p::Union{MDP,POMDP}, s, a, sp)
```

**Review by zsunberg — COMMENTED** [review]
I think we should definitely have this! Thanks for starting it.

Here are my concerns though:
1)  The info is created separately from the state transition. What if the info is something from deep inside the transition function, like intermediate or exceptional states that are reached during the transition calculation? In this PR, that information would have to be constructed after the fact from s and sp.

    Perhaps we need to have `generate_sri` and `generate_sori` functions in this package with default implementations like
    ```julia
    function generate_sri(p::Union{POMDP,MDP}, s, a, rng::AbstractRNG)
        sp, r = generate_sr(p, s, a, rng)
        return (sp, r, nothing) # nothing is the default info
    end
    ```
    That way, *if the problem writer wants to*, they can output info from anywhere in the calculation.

2) If the info function is independent, I don't think `generate_info` is the right name. The other `generate` functions all involve sampling a random outcome, and this one doesn't. Perhaps `info`, `get_info`, or `step_info` would be appropriate?

**Comment by zsunberg** [comment]
I wonder if we should have added a way to extract info from a step as part of the core POMDPs.jl package...

**Comment by rejuvyesh** [comment]
@zsunberg Those were precisely my concerns as well. On the one hand, `info` is supposed to be useful for debugging and similar things. Should they be part of the core api? I think yes.

**Comment by MaximeBouton** [comment]
is this redundant with #9 ?

**Comment by rejuvyesh** [comment]
It's better to merge them separately.

**Pull request closed without merging**